rustls/tests/api.rs

```diff
@@ -3432,8 +3432,8 @@ mod test_quic {
             let client_early = client.zero_rtt_keys().unwrap();
             let server_early = server.zero_rtt_keys().unwrap();
             assert!(equal_packet_keys(
-                &client_early.packet,
-                &server_early.packet
+                &*client_early.packet,
```

Comment:
ditto RE: `as_ref` here and 3519/3520. I won't leave the same feedback in each place but it's something to consider.